backend/src/models/user.ts

```diff
@@ -0,0 +1,30 @@
+import { ObjectId } from "mongodb";
+
+export default class User {
+  constructor(
+    public firstName: string,
+    public lastName: string,
+    public email: string,
+    public password: string,
+
+    public courses: string[],
+    public futureCourses: string[],
+    public hobbies: string[],
+    public languages: string[],
+    public programmingLanguages: string[],
+    public gender: string,
+    public age: number,
+    public wam: string,
+    public academicSocialRatio: number,
+
+    public preferredCourses: string[],
+    public preferredLanguages: string[],
+    public preferredProgrammingLanguages: string[],
+    public preferredGenders: string[],
```

Comment:
**question (non-blocking):** would we want to also enforce some sort of enum on these since we have pre-set values already?

Reply:
Coming soon?